src/Hadolint/Process.hs

```diff
@@ -64,21 +74,27 @@ import qualified Hadolint.Shell as Shell
 import Language.Docker.Syntax
 
 -- | Contains the required parameters for optional rules
-newtype RulesConfig = RulesConfig
+data RulesConfig = RulesConfig
   { -- | The docker registries that are allowed in FROM
-    allowedRegistries :: Set.Set Registry
+    allowedRegistries :: Set.Set Registry,
+    labelSchema :: Map.Map Text.Text Text.Text,
```

Comment:
I see. I've decided to go the simple route for now and just make it a `type LabelType = Text.Text`. This avoids having to do some sort of parsing.
The parsing is not desirable IMO, because we would need to parse both the input from the command line options as well as from the config file. But I'll go the extra mile if you insist that a custom `data` type is worth it here.